components/authz-service/server/policy.go

```diff
@@ -762,3 +764,32 @@ func (s *policyServer) logPolicies(policies []*storage.Policy) {
 	}
 	s.log.WithFields(kv).Info("Policy definition")
 }
+
+// requiredFieldsAndProjects verifies that all the required inputs are not empty
+func requiredFieldsAndProjects(obj interface {
+	GetId() string
+	GetName() string
+	GetProjects() []string
+}, resourceName string) error {
+	err := validate.RequiredIDandName(obj, resourceName)
+	if err != nil {
+		return err
+	}
+	return requiredProjects(obj.GetProjects())
+}
+
+// requiredProjects verifies that the projects are not empty and do not include '(unassigned)'
+func requiredProjects(projects []string) error {
```

Comment:
So I saw the conversation about moving this out of the lib into authz... then how to teams and tokens do project verification?

Reply:
Right now they're covered by this validation: https://github.com/chef/automate/blob/master/api/interservice/authz/v2/authz.proto#L94-L98

That works for now. But for consistency, we'll need to add this block of project validation into the ValidateProjectAssignment server code as well.

Reply:
We don't need to move it-- there's a case that this is the "authz validation lib" and it's meant to be used to validate yer things that use projects 🤷 . I was just thinking out loud 😬